### PR TITLE
[SPARK-12365][CORE] Use ShutdownHookManager where Runtime.getRuntime.addShutdownHook() is called

### PR DESCRIPTION
**Pull request by ted-yu**
SPARK-9886 fixed ExternalBlockStore.scala

This PR fixes the remaining references to Runtime.getRuntime.addShutdownHook()

### CODE REVIEW

**Comment by ted-yu** [comment]
Jenkins, test this please

**Comment by srowen** [comment]
Looks pretty good

**Comment by ted-yu** [comment]
@srowen 
Thanks for the quick review

**Comment by ted-yu** [comment]
For build 47811:
```
Had test failures in pyspark.mllib.tests with python2.6; see logs.
```
I don't think the above was related to this PR.

**Comment by ted-yu** [comment]
```
org.apache.spark.streaming.JavaAPISuite.testStreamingContextTransform

Failing for the past 1 build (Since Failed#47817 )
Took 62 ms.
Error Message

org.apache.spark.SparkException: Only one SparkContext may be running in this JVM (see SPARK-2243). To ignore this error, set spark.driver.allowMultipleContexts = true. The currently running SparkContext was created at: org.apache.spark.SparkContext$.getOrCreate(SparkContext.scala:2287) org.apache.spark.streaming.StreamingContext.<init>(StreamingContext.scala:140) org.apache.spark.streaming.StreamingContext$$anonfun$getOrCreate$1.apply(StreamingContext.scala:869) org.apache.spark.streaming.StreamingContext$$anonfun$getOrCreate$1.apply(StreamingContext.scala:869) scala.Option.map(Option.scala:145) org.apache.spark.streaming.StreamingContext$.getOrCreate(StreamingContext.scala:869) org.apache.spark.streaming.MasterFailureTest$.runStreams(MasterFailureTest.scala:278) org.apache.spark.streaming.MasterFailureTest$.testOperation(MasterFailureTest.scala:165) 
```
I don't think the above is caused by this PR.

**Comment by ted-yu** [comment]
Jenkins, test this please

File: core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala [hunk 1]
```diff
@@ -24,6 +24,7 @@ import org.apache.spark.deploy.rest.mesos.MesosRestServer
 import org.apache.spark.scheduler.cluster.mesos._
 import org.apache.spark.util.SignalLogger
 import org.apache.spark.{Logging, SecurityManager, SparkConf}
+import org.apache.spark.util.ShutdownHookManager
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala]: can you merge this with the on in L25

**Comment by ted-yu** [comment]
```
 > git fetch --tags --progress https://github.com/apache/spark.git +refs/pull/10325/*:refs/remotes/origin/pr/10325/* # timeout=15
ERROR: Timeout after 15 minutes
ERROR: Error fetching remote repo 'origin'
```

**Comment by ted-yu** [comment]
Jenkins, test this please

**Comment by andrewor14** [comment]
Merging into master 1.6

**Pull request closed without merging**